lib/recursive-methods.rb

```diff
@@ -1,35 +1,74 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
-
-# Time complexity: ?
-# Space complexity: ?
+require "pry"
+# Time complexity: O(n)
+# Space complexity: O(n)
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+  raise ArgumentError, 'Must provide a number greater than 1' unless n >= 0
+
+  if n == 1 || n == 0
+    return 1
+  else
+    return n * factorial(n - 1)
+  end 
+  
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse(s)
```

Comment:
👍 
This works, but because you create a new array with each recursive call this is O(n<sup>2</sup>) for both time/space complexity.